### ISSUE TITLE
Files are lost after moving them into a remote folder not encrypted as recipient with encryption enabled.

### ISSUE THREAD

**Issue by SergioBertolinSG** [owncloud/core#16510]
### Steps to reproduce

Having two servers serverA and serverB:
1. Enable encryption app in serverA.
2. Enable encryption in admin page of serverA.
3. Logout.
4. Login in serverB and don't enable encryption.
5. Share a folder REMOTE_FOLDER from admin in serverB to admin in serverA.
6. Login as admin in serverA.
7. Accept the federated shared folder from serverB.
8. Upload a file in root folder of serverA.
9. Move that file into REMOTE_FOLDER.
10. Enter into REMOTE_FOLDER, open the file.
11. Try to move the file back to the root folder.

### Expected behaviour
Files are moved into the REMOTE_FOLDER without problems.

### Actual behaviour
Files moved are lost. You can't access the file in step 10 and you can't move it back in step 11.

### Server configuration
**Operating system**:
Ubuntu 14.04

**Web server:**
Apache

**Database:**
MySQL

**PHP version:**
5.5.9

**ownCloud version:** (see ownCloud admin page)
{"installed":true,"maintenance":false,"version":"8.1.0.6","versionstring":"8.1 beta 2","edition":"Enterprise"}

**Updated from an older ownCloud or fresh install:**
Fresh

**List of activated apps:**
```

```

**The content of config/config.php:**

```

```

**Are you using external storage, if yes which one:** local/smb/sftp/...
No

**Are you using encryption:** 
Yes in serverA, no in serverB

**Logs**
```

```

### Client configuration

**browser**
Chrome 41, Firefox

**Comment by PVince81** [comment]
Possibly duplicate of https://github.com/owncloud/core/issues/13864

**Comment by schiesbn** [comment]
fixed with https://github.com/owncloud/core/pull/16512

 @SergioBertolinSG can you verify that the pr mentioned above fixes the issue? Thanks!

**Comment by PVince81** [comment]
@schiesbn that PR still needs fixing, see https://github.com/owncloud/core/pull/16512/files#r30872327

**Comment by schiesbn** [comment]
I know... But this are minor issues. The current state should already demonstrate that it fixes the issue... tested by my own.

**Comment by cmonteroluque** [comment]
@PVince81 @schiesbn 16512 seems to be a fairly small change. Can we agree on whether to apply it or not and close this one?

**Comment by schiesbn** [comment]
I think we will definitely apply it once it is ready (see my last comments). As soon as the pr is merged this issue will be closed automatically.

**Comment by SergioBertolinSG** [comment]
Reviewed in 8.1.0RC2. Working fine.
